WP3/D3.8/report.tex

```diff
@@ -185,13 +185,94 @@ \section{Continuous integration achievements of OpenDreamKit projects}\label{pro
 
 \subsection{SageMath}
 
-% Julian should remark on his efforts here: In particular what problems
-% motivated his efforts, and how he thinks this will improve things in the
-% future.  Maybe make note here about producing Docker images as build
-% artifacts (TODO: confirm if GAP is doing this too), and further how
-% mybinder.org turns out to be a useful development tool in conjunction
-% with Dockerized builds.  Not sure how much to say about that here, or
-% leave it to the conclusion.
+The SageMath project had traditionally employed two distinct CI systems, the
+Buildbot and Sage's own development, the
+patchbot\footnote{\url{https://github.com/sagemath/sage-patchbot}}.
+
+\subsubsection{Buildbot and patchbot}
+Sage's Buildbot runs as a non-advisory CI system on a wide variety of
+platforms. It builds Sage and runs its test suite for every change that gets
+accepted into the mainline of the Sage code. Its main purpose is not to be an
+advisory tool for developers but to catch problems and incompatibilities of
+Sage or its dependencies, in particular on more obscure platforms that
+dependent projects do often not test for. Due to the vast range of platforms
+covered, the Buildbot can not realistically be turned into an advisory CI
+system as it is not conceivable to provision and administer the number of
+machines that would be necessary to run Sage's test suite at every stage of a
+test that gets proposed by a developer.
+
+Therefore, the Sage community at some point developed its own patchbot system
```

Comment:
Here I would put "patchbot" in quotes (or maybe even Title Case) since it's a unique name of a product we developed.

Reply:
Sounds good to me.